internal/app/cache/cache.go

```diff
@@ -17,3 +22,112 @@ type Cache interface {
 	// AddWatch adds the watch to the cache and returns whether a stream is open.
 	AddWatch(key string, req envoy_api_v2.DiscoveryRequest) (bool, error)
 }
+
+type cache struct {
+	cache         *ristretto.Cache
+	expireSeconds time.Duration
+}
+
+type resource struct {
+	resp       *envoy_api_v2.DiscoveryResponse
+	watches    []*envoy_api_v2.DiscoveryRequest
```

Comment:
I think the watches are go channels

Reply:
Are we replacing watchers with response channels (https://envoyproxy.slack.com/archives/CTULFMN91/p1584409594031100) instead of DiscoveryRequests? This seems like a deviation from the cache interface, unless there's a way to get the DiscoveryRequest from its channel.

Reply:
sorry, my point was to call it something else, instead of watches.

Reply:
Gotcha. I think the spec had actually called them watchers instead of watches (which I'd missed in my initial PR) but since the terms are used interchangeably and might lead to confusion, I'm just going to rename this to requests.